metrics/sinks/influxdb/influxdb.go

```diff
@@ -257,17 +257,48 @@ func (sink *influxdbSink) createDatabase() error {
 	if sink.dbExists {
 		return nil
 	}
+
+	//query databases and check the specific database is exist or not.
 	q := influxdb.Query{
-		Command: fmt.Sprintf(`CREATE DATABASE %s WITH NAME "default"`, sink.c.DbName),
+		Command: "SHOW DATABASES",
 	}
 
-	if resp, err := sink.client.Query(q); err != nil {
-		if !(resp != nil && resp.Err != nil && strings.Contains(resp.Err.Error(), "already exists")) {
-			err := sink.createRetentionPolicy()
-			if err != nil {
-				return err
+	resp, err := sink.client.Query(q)
+
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Results) == 1 && len(resp.Results[0].Series) == 1 {
```

Comment:
Check whether `resp` is `nil` like https://github.com/kubernetes/heapster/pull/2086/files#diff-a83c1db1f245f8fb4c41e99889717525L265